pkg/kube/logs.go

```diff
@@ -31,6 +31,9 @@ func (r *logsReader) GetLogsByJobAndContainerName(ctx context.Context, job *batc
 	if err != nil {
 		return nil, fmt.Errorf("getting pod controllered by job: %q: %w", job.Namespace+"/"+job.Name, err)
 	}
+	if pod == nil {
```

Comment:
my bad the `==` and `!=` is rendered similar on my editor, missed editing this after testing different flows.